api/admin/controller/library_settings.py

```diff
@@ -316,13 +320,15 @@ def list_setting(self, setting, json_objects=False):
         else:
             # Allow any entered values.
             value = []
-            inputs = flask.request.form.getlist(setting.get("key"))
-            for i in inputs:
-                if not isinstance(i, list):
-                    i = [i]
-                if json_objects:
-                    i = [json.loads(x) for x in i]
-                value.extend(i)
+            inputs = flask.request.form.getlist(setting.get("key")) if setting.get("type") == "list" else flask.request.form.get(setting.get("key"))
```

Comment:
Reformat this as a four-line if/else statement. This is tough to read, and since you're not in a list comprehension it doesn't all have to be one statement.